test/api/pegjs-api.spec.js

```diff
@@ -206,7 +206,46 @@ describe("Peggy API", () => {
         }
       });
 
-      describe("does not throws an exception on reserved JS words used as a rule name", () => {
+      describe("does not throw an exception on special non-reserved JS words used as a label", () => {
+        const nonReserved = [
+          // Reserved in old ECMAScript versions
+          "abstract",
+          "boolean",
+          "byte",
+          "char",
+          "double",
+          "final",
+          "float",
+          "goto",
+          "int",
+          "long",
+          "native",
+          "short",
+          "synchronized",
+          "throws",
+          "transient",
+          "volatile",
+          // Never reserved, but used in special language constructs
+          "as",
+          "async",
+          "from",
+          "get",
+          "of",
+          "set",
+        ];
+        for (const label of nonReserved) {
+          it(label, () => {
+            expect(() => {
```

Comment:
I don't think you need to use expect().to.not.throw() here.  You could double check by temporarily adding a real reserved word to ensure that the test fails.

Can you please use the label in the grammar, like this:

```js
peg.generate([
  "start = " + label + ":end { return " + label + "; }",
  "end = 'a'",
].join("\n"));
```

Also, please execute the generated parse function to ensure that it works at runtime as well.

Reply:
Indeed. The `to.not.throw` is completely unnecessary. Was just following an example set by existing test.